packages/turbo-repository/rust/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+use anyhow::{anyhow, Result};
+use napi_derive::napi;
+use turbopath::AbsoluteSystemPathBuf;
+use turborepo_repository::{
+    inference::{RepoMode, RepoState},
+    package_manager::PackageManager as RustPackageManager,
+};
+
+#[napi]
+pub struct Repository {
+    repo_state: RepoState,
+    pub root: String,
+    pub is_monorepo: bool,
+}
+
+#[napi]
+pub struct PackageManager {
+    #[allow(dead_code)]
+    package_manager: RustPackageManager,
+    pub name: String,
+}
+
+impl From<RustPackageManager> for PackageManager {
+    fn from(package_manager: RustPackageManager) -> Self {
+        Self {
+            name: package_manager.to_string(),
+            package_manager,
+        }
+    }
+}
+
+#[napi]
+impl Repository {
+    #[napi(factory, js_name = "detectJS")]
+    pub fn detect_js(path: Option<String>) -> Result<Self> {
```

Comment:
I'd prefer to anchor on "web" rather than "JS" as our ecosystem (e.g. a Next.js app isn't a JS app, it's a _web_ app)

Reply:
Mentioned elsewhere, but I don't think that's correct. Our ecosystem is not the (only) web, but anything JS, including backend services and CLIs.